### PR TITLE
9721 - document qc links bugfix to test

### PR DESCRIPTION
**Pull request by jasonfu9966**
https://app.zenhub.com/workspaces/flexionef-cms-5bbe4bed4b5806bc2bec65d3/issues/flexion/ef-cms/9721
testin time (for real this time)

### CODE REVIEW

**Review by TomElliottFlexion — COMMENTED** [review]
Should we consider trying to add an integration test?

**Review by ujahio — APPROVED** [review]

**Review by shumway-tm — APPROVED** [review]

File: shared/src/business/useCases/docketEntry/editPaperFilingInteractor.ts [hunk 1]
```diff
@@ -56,6 +56,16 @@ export const editPaperFilingInteractor = async (
     docketEntryId: primaryDocumentFileId,
   });
 
+  let error;
+  if (!currentDocketEntry) {
+    error = new NotFoundError('Docket entry not found');
+  } else if (currentDocketEntry.servedAt) {
+    error = new Error('Docket entry has already been served');
+  }
+  if (error) {
```
Comment (rachelschneiderman) [inline shared/src/business/useCases/docketEntry/editPaperFilingInteractor.ts]: Can this be condensed to just throw inside the above if statements?

Comment (jasonfu9966) [reply shared/src/business/useCases/docketEntry/editPaperFilingInteractor.ts]: Good catch! I made the change.

File: shared/src/business/useCases/courtIssuedDocument/fileAndServeCourtIssuedDocumentInteractor.ts [hunk 2]
```diff
@@ -71,25 +71,37 @@ export const fileAndServeCourtIssuedDocumentInteractor = async (
 
   const subjectCaseEntity = new Case(subjectCase, { applicationContext });
 
-  const docketEntryToServe = subjectCaseEntity.getDocketEntryById({
+  const originalSubjectDocketEntry = subjectCaseEntity.getDocketEntryById({
```
Comment (rachelschneiderman) [inline shared/src/business/useCases/courtIssuedDocument/fileAndServeCourtIssuedDocumentInteractor.ts]: 📓 Just a note that this was renamed from `originalSubjectDocketEntry` to `docketEntryToServe` as part of the #9681 refactor, this branch may encounter a merge conflict with staging if that story is merged first.

File: shared/src/business/useCases/docketEntry/editPaperFilingInteractor.test.ts [hunk 3]
```diff
@@ -83,6 +100,31 @@ describe('editPaperFilingInteractor', () => {
     ).rejects.toThrow('Unauthorized');
   });
 
+  it('should throw an error if docket entry is not valid', async () => {
```
Comment (rachelschneiderman) [inline shared/src/business/useCases/docketEntry/editPaperFilingInteractor.test.ts]: This test name does not seem to match the expectation

File: web-client/src/providers/socketRouter.js [hunk 4]
```diff
@@ -88,6 +88,11 @@ export const socketRouter = (app, onMessageCallbackFn) => {
           ...message,
         });
         break;
+      case 'serve_document_complete_error':
```
Comment (rachelschneiderman) [inline web-client/src/providers/socketRouter.js]: Nit: Should this just be `serve_document_error`? As most likely service never completed if there was an error

File: web-client/src/presenter/sequences/serveDocumentErrorSequence.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import { clearModalStateAction } from '../actions/clearModalStateAction';
+import { setShowModalFactoryAction } from '../actions/setShowModalFactoryAction';
+import { unsetWaitingForResponseAction } from '../actions/unsetWaitingForResponseAction';
+
+export const serveDocumentErrorSequence = [
```
Comment (rachelschneiderman) [inline web-client/src/presenter/sequences/serveDocumentErrorSequence.js:5]: 👏 Love the name of this sequence

File: web-client/src/presenter/sequences/serveDocumentErrorSequence.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+  unsetWaitingForResponseAction,
+  clearModalStateAction,
+  setShowModalFactoryAction('WorkItemAlreadyCompletedModal'),
```
Comment (rachelschneiderman) [inline web-client/src/presenter/sequences/serveDocumentErrorSequence.js]: 💭 What would y'all think about naming the modal using a similar noun to the sequence? I found it a tad confusing that the sequence is speaking in terms of a `document` and then the modal that opens has to do with a `work item`.  Just a thought, let me know what you think!

File: shared/src/business/useCases/courtIssuedDocument/fileAndServeCourtIssuedDocumentInteractor.ts (hunk 2 above)
Comment (jasonfu9966) [reply shared/src/business/useCases/courtIssuedDocument/fileAndServeCourtIssuedDocumentInteractor.ts]: yeah, I caught that with the unit test. I'll let y'all merge first and fix this after.

**Review by codyseibert — APPROVED** [review]

File: web-client/src/presenter/sequences/serveDocumentErrorSequence.js (hunk 6 above)
Comment (jasonfu9966) [reply web-client/src/presenter/sequences/serveDocumentErrorSequence.js]: made a little change that gets around this fact--let me know if it works for ya.

Comment (rachelschneiderman) [reply web-client/src/presenter/sequences/serveDocumentErrorSequence.js]: Thanks! I think I was referring more to renaming the sequence to say `workItemAlreadyCompletedErrorSequence` OR renaming the modal to be called `ServeDocumentErrorModal`, if that makes sense. I'm OK with merging this as-is, it was just a thought 😄 Thank you!

**Review by rachelschneiderman — APPROVED** [review]

**Pull request merged**